### ISSUE TITLE
Better support for SD cards in Android 4.4 and newer

### ISSUE THREAD

**Issue by arnowelzel** [abrensch/brouter#169]
Android 4.4 and newer does not allow to write to an external SD card directly. However it is still possible using the storage access framework and also using app private folders on the SD card.

It would be nice, if BRouter would support either a private app folder on the SD card or the storage access framework (which involves requesting permission by the user for accessing the external storage).

**Comment by abrensch** [comment]
see also here: https://github.com/abrensch/brouter/issues/139

so I have to do something to become back able to update on Google-Play, and my current plan is:

- on >= 4.4, the base-directory dialogs offers the 2 choices for the Android/data/... directories for int/ext card. That implies that downloaded rd5's are deleted on app-uninstall, but that's the way it is

- on >=6, reading waypoint-databases from maptools is no longer possible with the Google-Play version. The downlodable release APK will be build against target-API 28 and behave as before.  I think, using BRouter in "app-mode" becomes less and less important, so users asking for the appmode are also willing to install the release-apk.

**Comment by abrensch** [comment]
see version 1.5.5 -> you get a base-dir proposal on external sd (Android/data/... dir)

**Comment by arnowelzel** [comment]
Works perfectly on my device with Android 8.1. Thank you!

The only (little) downside: I had BRouter installed already and after the update it did forget the previous folder location and asked me again which folder to use. I stopped at the point where I had to select the areas to download, left BRouter and manually move the downloaded files from my custom folder on the SD card to the private app folder. So the data was usable again. However users with less experience might end up with downloading everything again and having multiple copies of the files left.
